hbase_sql/src/main/java/com/splicemachine/derby/stream/function/BulkLoadKVPairFunction.java

```diff
@@ -35,10 +40,20 @@ public BulkLoadKVPairFunction(long conglomerateId) {
     }
 
     @Override
-    public Tuple2<Long,Tuple2<byte[], byte[]>> call(KVPair kvPair) throws Exception {
+    public Iterator<Tuple2<Long, Tuple2<byte[], byte[]>>> call(Iterator<KVPair> kvPairIterator) throws Exception {
+        if (!kvPairIterator.hasNext())
+            return Collections.EMPTY_LIST.iterator();
 
-        Tuple2<byte[], byte[]> kv = new Tuple2(kvPair.getRowKey(), kvPair.getValue());
-        return new Tuple2(conglomerateId, kv);
+        List<Tuple2<Long, Tuple2<byte[], byte[]>>> outList = new ArrayList<>();
+        while (kvPairIterator.hasNext()) {
+            KVPair kvPair = kvPairIterator.next();
+            if (kvPair == null)
+                continue;
+            Tuple2<byte[], byte[]> kv = new Tuple2(kvPair.getRowKey(), kvPair.getValue());
+            outList.add(new Tuple2(conglomerateId, kv));
+        }
+
+        return outList.iterator();
```

Comment:
Are you meaning to materialize the whole iterator into a list?  Is it not better to wrap the iterator vs. materializing the whole thing?

Reply:
Ok, looked at this.  I do not think you can do it this way.  This would take an iterator for the entire partition and materialize it into a list (expensive).

Reply:
Thanks @jleach4 for the comment. I pushed a commit to implement an iterator instead of materializing all rows into a list.